### PR TITLE
[Model Enabling] Support ChatGLM3

### PR DESCRIPTION
**Pull request by Zhenzhong1**
## Type of Change

New Feature.

## Description

Support ChatGLM3 in the Neural Speed.

## Expected Behavior & Potential Risk

N/A

## How has this PR been tested?

numactl -m 0 -C 0-55 python scripts/run.py /home/zhenzhong/model/chatglm3-6b/ -p "你好" --model_type=chatglm3
image

## Dependency Change?

N/A

### CODE REVIEW

File: neural_speed/application/main_run.cpp [hunk 1]
```diff
@@ -238,9 +238,12 @@ int main(int argc, char** argv) {  // NOLINT
     std::vector<std::string> prompts;
     prompts.push_back(params.prompt);
     std::string prompt = build_prompt_glm2(prompts);
+    std::cout << "prompt = " << prompt << std::endl;
+
```
Comment (a32543254) [inline neural_speed/application/main_run.cpp:242]: why we need cout here?

**Review by a32543254 — APPROVED** [review]
LGTM

**Comment by a32543254** [comment]
Could you also add extension test, and post the benchmark data here?

**Review by intellinjun — APPROVED** [review]

File: neural_speed/application/main_run.cpp (hunk 1 above)
Comment (Zhenzhong1) [reply neural_speed/application/main_run.cpp]: fixed

**Comment by a32543254** [comment]
> Could you also add extension test, and post the benchmark data here?

No need, seems chat GLM3 share same structure with chat GLM2. so we can treat them as one.

**Comment by Zhenzhong1** [comment]
> > Could you also add extension test, and post the benchmark data here?
> 
> No need, seems chat GLM3 share same structure with chat GLM2. so we can treat them as one.

yes. It's true. It's ok. I have added it. It also helps us to check convert / queantize / inference pipeline.

https://github.com/intel-innersource/frameworks.ai.lpot.lpot-validation/pull/623/files

File: docs/prompt_template.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+prompt = "你好"
+tokenizer = AutoTokenizer.from_pretrained(args.model_path, trust_remote_code=True)
+inputs = tokenizer.build_chat_input(prompt)['input_ids']
+model = Model()
+model.init_from_bin(args.model_name, gguf_path)
+outputs = model.generate(inputs, max_new_tokens=300, do_sample=True)
+words = tokenizer.decode(outputs[0])
+```
+
+## Chat with LLaMA2:
```
Comment (a32543254) [inline docs/prompt_template.md:21]: LLaMA2   
>>
LLaMa2

why capitalize second A ?

Comment (Zhenzhong1) [reply docs/prompt_template.md:21]: don't know, copied from other place.

Comment (Zhenzhong1) [reply docs/prompt_template.md:21]: It may have appeared somewhere. OH. Here.

https://github.com/intel/neural-speed/blob/main/docs/supported_models.md

**Pull request merged**